src/Runtime/OMInstrument.inc

```diff
@@ -55,6 +59,9 @@ void OMInstrumentInit() {
 	  virtualMemEnabled = !virtualMemEnabled;
   }
 
+#ifdef _WIN32
```

Comment:
Rather than having ifdefs in the functions, can you create a couple of helper functions and/or macros to encapsulate the functionality that will be different and then just call them? That will make the code easier to read and maintain.

Reply:
Made the change. I changed the semantics of reporting memory in case threads are forked along the execution.